src/snowflake/connector/connection_diagnostic.py

```diff
@@ -214,6 +214,12 @@ def __test_socket_get_cert(
                 context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
                 sock = context.wrap_socket(conn, server_hostname=host)
                 certificate = ssl.DER_cert_to_PEM_cert(sock.getpeercert(True))
+                http_request = f"""GET / {host}:{port} HTTP/1.1\r\n
```

Comment:
is this method `__test_socket_get_cert` being tested?

Reply:
This is tested indirectly with the http_host_report test, but added a direct test for this as well.